app/scripts/metamask-controller.js

```diff
@@ -2399,6 +2399,9 @@ export default class MetamaskController extends EventEmitter {
         allowedEvents: [],
       }),
       disabled: !this.preferencesController.state.useExternalServices,
+      getMetaMetricsId: Promise.resolve(
```

Comment:
Why is `Promise.resolve` used here?

Reply:
Oh, maybe you meant this to be

```suggestion
      getMetaMetricsId: async () => this.metaMetricsController.getMetaMetricsId()
```

Right now it appears to be broken, as it's passing in a `Promise<string>` rather than `() => Promise<string>`

We could also consider simplifying the type in core to allow this to be asynchronous or synchronous (e.g. `() => Promise<string> | string` so we don't need the `async () => { ... }` wrapper here, but that's not super important.

Reply:
It actually works with directly passing in the function itself, modified in 29f38193a10871d06c68ff580a97001be4eff8d5

Reply:
It looks like you're still passing in a value, not a function

Reply:
In the end i decided to modify the controller side and pass in a synchronous function from extension side 🙏🏻 will catch in next package of `@metamask/remote-feature-flag-controller`